src/runtime/module_jit_ref_count.cpp

```diff
@@ -0,0 +1,29 @@
+#include "./runtime_internal.h"
+
+extern "C" {
+
+WEAK void *halide_jit_module_argument = NULL;
+WEAK void (*halide_jit_module_adjust_ref_count)(void *arg, int32_t count) = NULL;
+
+}
+
+namespace Halide { namespace Runtime { namespace Internal {
+
+WEAK void halide_use_jit_module() {
```

Comment:
This occurred to me this morning: When a device buffer is freed, the reference count on the containing module is decremented, potentially deleting the currently-executing function (unless we can guarantee tail recursion all the way back out of the runtime)?

Looks like it's not a problem for Buffer, because it currently holds an independent handle to the JITModule. But in the general case this seems problematic. Either the owner of the buffer_t needs to do incref() free_dev_buffer() decref(), or equivalently manage the reference count itself, or we need a way to delay the destruction until we escape the runtime.